lib/grinder.dart

```diff
@@ -317,10 +317,9 @@ class Grinder {
     if (targets.isEmpty) {
       if (defaultTask != null) {
         targets = [defaultTask.name];
-      } else {
-        if (!dontRun) log('no tasks specified, and no default task defined');
+      } else if (!dontRun) {
+        throw new GrinderException('Tried to call non-existent default task.');
```

Comment:
I moved the error to the cli-specific code:

https://github.com/google/grinder.dart/pull/213/files#diff-48f50be87f4906250da81d6ab77b472aR60

So the cli will catch this before `Grinder.start` is even called.  But I still kept an exception here to cover programmatic usage of `Grinder.start`.